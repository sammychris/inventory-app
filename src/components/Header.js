import React from 'react';
import styled from 'styled-components';
// import { Link } from 'react-router-dom';
// import closeIcon from '../assets/close-icon.svg';
// import cartIcon from '../assets/cart-icon.svg';
// import CartOverlay from './CartOverlay';
// import CurrencySwitcher from '../components/CurrencySwitcher';
// import Logo from '../components/Logo';
// import withHeader from '../hoc/withHeader';

const Header = () => {
    return (
      <>
        <Container>
          <HeaderSection>
            <Navigation>
              {/* {
                data?.categories?.map((el, i) => <NavLink to={`/category/${el.name}`} key={i}>{el.name}</NavLink>)
              } */}
              Logo
            </Navigation>
            {/* <Logo /> */}
            <Actions>
              {/* <Currency ref={currencyIconRef} href="#">
                {currency.symbol}
                <SelectCurrency src={closeIcon} />
              </Currency> */}
              {/* <CartIcon href="#" ref={cartIconRef}>
                <ItemNumber>
                  {
                    cart.length ? cart?.map((el) => el.quantity)?.reduce((a, b) => a + b) : 0
                  }
                </ItemNumber>
              </CartIcon> */}
              User
            </Actions>
            {/* {showCart && <CartOverlay showCartRef={showCartRef} onViewCart={onViewCart} />} */}
            {/* {showCurrency && (
              <CurrencySwitcher
                currencies={data?.currencies}
                currency={currency}
                onClick={onChangeCurrency}
                showCurrencyRef={showCurrencyRef}
              />
            )} */}
          </HeaderSection>
        </Container>
        {/* {showCart && <Overlay />} */}
      </>
    );
  }


export default Header;

const Container = styled.div`
  position: relative;
  z-index: 2;
  background: #fff;
`;

const HeaderSection = styled.header`
  height: 80px;
  max-width: 1230px;
  width: 90%;
  margin: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: relative;
`;

const Overlay = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  background: rgb(0 0 0 / 42%); 
  z-index: 1;
`;

const Navigation = styled.div`
  display: flex;
  text-align: center;
  align-items: center;
  height: 100%;
  width: 234px;
  justify-content: space-around;
`;

// const NavLink = styled(Link)`
//   font-size: 16px;
//   color: #1D1F22;
//   text-decoration: none;
//   text-transform: uppercase;
// `;

const Actions = styled.div`
  width: 234px;
  display: flex;
  justify-content: flex-end;
  position: relative;
`;

const Currency = styled.a`
  margin-right: 22px;
  display: flex;
  width: 25px;
  justify-content: space-between;
  align-items: center;
  color: #1D1F22;
  text-decoration: none;
`;

const SelectCurrency = styled.img`
  width: 6px;
  height: 3px;
`;

// const CartIcon = styled.a`
//   width: 20px;
//   height: 20px;
//   background: url(${cartIcon});
// `;

const ItemNumber = styled.span`
  background: #1D1F22;
  border-radius: 50%;
  font-family: roboto;
  font-weight: 700;
  font-size: 14px;
  color: #fff;
  padding: 2px 6px;
  position: absolute;
  top: -10px;
  right: -10px;
`;