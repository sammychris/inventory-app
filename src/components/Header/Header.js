import React from 'react';
import styled from 'styled-components';

const Header = () => {
    return (
      <>
        <Container>
          <HeaderSection>
            <Navigation>
              Logo
            </Navigation>
            <Actions>
              User
            </Actions>
          </HeaderSection>
        </Container>
      </>
    );
  }


export default Header;

const Container = styled.div`
  position: relative;
  z-index: 2;
  background: #fff;
`;

const HeaderSection = styled.header`
  height: 80px;
  max-width: 1230px;
  width: 90%;
  margin: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: relative;
`;

const Navigation = styled.div`
  display: flex;
  text-align: center;
  align-items: center;
  height: 100%;
  width: 234px;
  justify-content: space-around;
`;

const Actions = styled.div`
  width: 234px;
  display: flex;
  justify-content: flex-end;
  position: relative;
`;