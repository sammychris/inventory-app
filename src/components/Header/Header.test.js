import { shallow } from "enzyme";
import Header from "./Header";

const wrapper = shallow(<Header />);

test("expect to render a Header", () => {
  expect(wrapper.length).toEqual(1);
});

test("expect to make a snap shot to Header component", () => {
  expect(wrapper).toMatchSnapshot();
});