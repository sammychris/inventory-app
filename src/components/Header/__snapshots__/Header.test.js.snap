// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to make a snap shot to Header component 1`] = `ShallowWrapper {}`;
